apps/mobile/src/components/DrawerNavigation/Sidebar.tsx

```diff
@@ -0,0 +1,54 @@
+import type { FC } from 'react';
+import { View } from 'react-native';
+import type { DrawerContentComponentProps } from '@react-navigation/drawer';
+import DashboardNavigator from '@walless/app/features/DashboardLayout/Navigator';
+import {
+	useSafeAreaInsets,
+	useSnapshot,
+	useWidgets,
+} from '@walless/app/utils/hooks';
+import { appState, mockWidgets } from '@walless/engine';
+import { modules } from '@walless/ioc';
+import type { WidgetDocument } from '@walless/store';
+import { appActions } from 'state/app';
+
+export const sidebarWidth = 64;
+
+const IS_HARDCODED = true;
+
+export const Sidebar: FC<DrawerContentComponentProps> = ({ navigation }) => {
+	const { profile, activeWidgetId } = useSnapshot(appState);
+	const widgets = useWidgets();
+
+	const handleExtensionPress = (item: WidgetDocument) => {
+		const id = item._id === '' ? 'Explore' : item._id;
+		appActions.setActiveWidgetRoute(id);
+		navigation.navigate(id);
+	};
+
+	const handleRemoveLayout = async (layout: WidgetDocument) => {
+		await modules.storage.put(layout);
```

Comment:
I think we should avoid using storage methods directly, try to unify actions related to similar entity (i.e tokens, widgets) under one object, reference `appActions`, `tokenActions` etc.